processor/src/main/java/org/mapstruct/ap/internal/model/LifecycleCallbackFactory.java

```diff
@@ -99,16 +113,18 @@ private LifecycleCallbackFactory() {
             false) );
     }
 
-    private static List<LifecycleCallbackMethodReference> toLifecycleCallbackMethodRefs(
+    private static List<LifecycleCallbackMethodReference> toLifecycleCallbackMethodRefs(Method method,
             List<SourceMethod> candidates, Map<SourceMethod, List<Parameter>> parameterAssignmentsForSourceMethod,
-            MappingBuilderContext ctx) {
-
+            MappingBuilderContext ctx, Set<String> existingVariableNames) {
         List<LifecycleCallbackMethodReference> result = new ArrayList<LifecycleCallbackMethodReference>();
         for ( SourceMethod candidate : candidates ) {
             markMapperReferenceAsUsed( ctx.getMapperReferences(), candidate );
-            result.add( new LifecycleCallbackMethodReference(
-                candidate,
-                parameterAssignmentsForSourceMethod.get( candidate ) ) );
+            result.add(
+                new LifecycleCallbackMethodReference(
+                    candidate,
+                    parameterAssignmentsForSourceMethod.get( candidate ),
+                    method.getReturnType(),
```

Comment:
This needs to be `method.getResultType()` (that's also the type used to declare the result object type in the generated method).